Update ContainerCluster fields

fkc1e100 opened the pull request:
## Description
Adds support for missing fields in `ContainerCluster`:
- EnableDefaultComputeClass (mapped to defaultComputeClassEnabled)
- EnableK8sTokensViaDNS (mapped to enableK8sTokensViaDns)
- EnableCiliumClusterwideNetworkPolicy
- EnableFqdnNetworkPolicy

## Implementation
- Regenerated `ContainerCluster` Go types to include these fields from the Terraform provider.
- Verified via E2E test `TestAllInSeries/fixtures/containercluster-new-fields` against MockGCP.

justinsb commented on apis/apphub/v1alpha1/application_identity.go:
```diff
@@ -57,7 +57,7 @@ func NewApplicationIdentity(ctx context.Context, reader client.Reader, obj *AppH
 	// Get Parent
 	parent := obj.Spec.Parent
-	projectRef, err := refsv1beta1.ResolveProject(ctx, reader, obj.GetNamespace(), parent.ProjectRef)
+	projectRef, err := refsv1beta1.ResolveProject(ctx, reader, obj.GetNamespace(), parent.ProjectRef) //nolint:typecheck
```
Nit: surprised you need the nolint marker when we didn't before, but /shrug

justinsb commented on third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/container/resource_container_cluster.go:
```diff
@@ -1929,6 +1935,18 @@ func ResourceContainerCluster() *schema.Resource {
 				Description: `Whether FQDN Network Policy is enabled on this cluster.`,
 				Default:     false,
 			},
+			"enable_cilium_clusterwide_network_policy": {
```
I would also expect that we will need corresponding changes in the tf Update and Create methods (at least).

I think if we provided the latest TF code in the context, that would help Gemini (assuming TF has added these fields upstream)

fkc1e100 replied:
Thanks verifying this! I have gone through and added the explicit `Create`/`Read`/`Update` logic for `EnableK8sTokensViaDNS` (using `HAS_CHANGE`), and verified that `DefaultComputeClassEnabled` is handled correctly via the existing `expandManagement`/`flattenManagement` paths. I also ensured the NetworkPolicy fields (`EnableCiliumClusterwideNetworkPolicy`, `EnableFqdnNetworkPolicy`) are fully covered in the CRUD operations.
Relevant commits for implementation:
*   Initial CRD & Types: bd71b3f9
*   Provider Logic Mapping: efdcd7a4
*   Enablement & E2E Tests: f53fc4ce

maqiuyujoyce commented on pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-new-fields/create.yaml:
```diff
@@ -0,0 +1,23 @@
+apiVersion: container.cnrm.cloud.google.com/v1beta1
```
Could you try to run the test locally following this process: https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/mockgcp#capture-golden-object-and-http-golden-logs? With that you can verify before we trigger presubmit test.

fkc1e100 replied:
Done. I have re-run the tests following the MockGCP guidance. The golden HTTP logs (`_http.log`) have been captured and included in the latest commit. I also verified the test passes against both real GCP and MockGCP.

fkc1e100 commented:
**Summary of Resolution:**

Addressing the feedback regarding atomic updates, prompt improvements, and testing:

1.  **Process Update (Development Guide):** I have updated my local AI Development Guide (`README.md` - not currently in the repo) to explicitly enforce a **"One Field Per PR"** constraint for future tasks. This directly addresses the feedback about preferring atomic field additions to simplify reviews.
2.  **Code/Prompt Cleanliness:**
    *   Updated prompts to strictly forbid `//nolint` directives.
    *   Removed existing `//nolint` directives from this PR and verified that `make vet` passes cleanly.
3.  **Testing & Verification:**
    *   Fixed the `create.yaml` fixture to satisfy GKE requirements (`autoUpgrade: true` for `nodeAutoProvisioning` on REGULAR channel).
    *   Successfully ran E2E tests against **Real GCP** (cluster created and deleted) and **MockGCP**.
    *   Captured and committed the missing `_http.log` golden files.
4.  **Logic Fixes:** Ensured `EnableCiliumClusterwideNetworkPolicy` and other new fields are correctly mapped in `Create`, `Read`, and `Update` logic (handling hidden Terraform fields).

This PR is now ready for final review.

maqiuyujoyce reviewed (commented):
Thanks for the update, Frank! I approved the presubmit test run so that you can have a generic idea about what test needs to be verified before you do another round of update.

fkc1e100 commented:
This one has been a learning exercise but it's run its course. Learning is to not run four fields in parallel. WIll open separate Issues for each field.

fkc1e100 commented:
EnableDefaultComputeClass
https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/5900
EnableK8sTokensViaDNS
https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/6200
EnableCiliumClusterwideNetworkPolicy
https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/6201
EnableFqdnNetworkPolicy
https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/6202
